tests/data/sources/test_datasource.py

```diff
@@ -24,31 +26,54 @@ def ds_parser(**kwargs):
             )
 
         DataSource.add_supported_format("new-format", ds_parser)
-        DataSource.add_supported_format("new-format", ds_parser)
-        ds = DataSource(format="new-format")
-        self.assertFalse(ds.to_dataframe().columns is None)
+
+        for ds in [DataSource(format="new-format"), DataSource(source="new-format")]:
```

Comment:
I think we should not test deprecated behavior to free us from overload. That is, no need to test `DataSource(format="new-format")`.